### DISCUSSION TITLE
Make AMQP 1.0 and AMQP 0-9-1 connections optional

### DISCUSSION THREAD

**Discussion by johanrhodin** [rabbitmq/rabbitmq-server#12238] [Ideas] [answered]
With Native AMQP 1.0 in 4.0 it will no longer be possible to only disable AMQP 1.0 connections. (Which was possible in 3.x, since it was a plugin - Don't enable it, don't receive AMQP 1.0 traffic)

The suggestion here is to make it configurable whether to allow AMQP 1.0 and/or AMQP 0-9-1. Right now the only possibility would be to block port/not start the listener for both of them.

One idea is to introduce a check here: https://github.com/rabbitmq/rabbitmq-server/commit/8cb313d5a106ee23a7cbac98160d552500a15776#diff-c6da5f893cf5567c62329bbe62145d8dfd282a3f8ad4f00295459ed4e89c0de0L1650 and a similar check for AMQP 0-9-1.

**Comment by kjnilsson** [comment]
Why would you want to?

*Karl Nilsson*

**Reply by johanrhodin** [reply]
One is security: run the least amount of services required. Example https://github.com/advisories/GHSA-5hjq-8gx9-79g4, similar issues can be found for MQTT/STOMP (which can be disabled by not running the respective plugins).
Another is reducing complexity (be sure that an issue wasn't started by usage of this or that protocol).

**Reply by michaelklishin** [reply]
@johanrhodin why a solution to every problem should be a modification to RabbitMQ, that the core team will have to maintain (even if someone else contributes it initially), document, consider in support cases and automation tools?

Why can't you simply block traffic on certain ports? Your company fully controls the deployment process.

**Comment by michaelklishin** [comment, marked as answer]
99.99% of users do not need this. AMQP 1.0 is the future of RabbitMQ, AMQP 0-9-1 is not going away, if you want to disable both it should be enough to simply block traffic to the well known AMQP ports.

Other protocol plugins remain, well, plugins.
